subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -106,12 +174,16 @@ int dfu_target_mcuboot_write(const void *const buf, size_t len)
 	}
 
 	offset += len;
+	store_offset();
 
 	return 0;
 }
 
 int dfu_target_mcuboot_done(bool successful)
 {
+	offset = 0;
+	store_offset();
```

Comment:
Should we do the  `IS_ENABLED(CONFIG_SETTINGS_FCB)` check around the calls to `store_offset` instead of inside it? This way it is clearer that the offset is only stored when you have the correct configs.

Reply:
It will be harder to maintain since you would have to change at every occurrence of store_offset. Currently only two but still.

I do see that it will be a bit clear.